mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICALToJsonAttribute.java

```diff
@@ -170,27 +169,53 @@ public void service(Mail mail) throws MessagingException {
     }
 
     private void setAttribute(Mail mail, Map<String, Calendar> calendars, Map<String, byte[]> rawCalendars) throws MessagingException {
-        Optional<String> sender = retrieveSender(mail);
+        Optional<MailAddress> sender = retrieveSender(mail);
         if (!sender.isPresent()) {
             LOGGER.info("Skipping {} because no sender and no from", mail.getName());
             return;
         }
+
+        MailAddress transportSender = sender.get();
+        MailAddress replyTo = fetchReplyTo(mail).orElse(transportSender);
```

Comment:
I don't get why you need this orElse

Reply:
Because we should always have one and `replyTo` might not be present.